lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -2417,6 +2434,98 @@ selectSeqStatePendingIxs wid =
   where
     fromRes = fmap (W.Index . seqStatePendingIxIndex . entityVal)
 
+instance
+    ( PersistPublicKey (k 'AccountK)
+    , MkKeyFingerprint k (Proxy n, k 'AddressK XPub)
+    , PaymentAddress n k
+    , SoftDerivation k
+    , WalletKey k
+    , Typeable n
+    , Seq.GetPurpose k
+    ) => PersistState (Shared.SharedState n k) where
+    insertState (wid, sl) st = do
+        case st of
+            Shared.SharedState prefix (Shared.PendingFields (Shared.SharedStatePending accXPub pTemplate dTemplateM g)) -> do
+                insertSharedState accXPub g pTemplate dTemplateM prefix
+                insertCosigner (cosigners pTemplate) Payment
+                when (isJust dTemplateM) $
+                    insertCosigner (fromJust $ cosigners <$> dTemplateM) Delegation
+
+            Shared.SharedState prefix (Shared.ReadyFields pool) -> do
+                let (Seq.ParentContextMultisigScript accXPub pTemplate dTemplateM) =
+                        Seq.context pool
+                insertSharedState accXPub (Seq.gap pool) pTemplate dTemplateM prefix
+                insertCosigner (cosigners pTemplate) Payment
+                when (isJust dTemplateM) $
+                    insertCosigner (fromJust $ cosigners <$> dTemplateM) Delegation
+                insertAddressPool @n wid sl pool
+      where
+         insertSharedState accXPub g pTemplate dTemplateM prefix = do
+             deleteWhere [SharedStateWalletId ==. wid]
+
+             insert_ $ SharedState
+                 { sharedStateWalletId = wid
+                 , sharedStateAccountXPub = serializeXPub accXPub
+                 , sharedStateScriptGap = g
+                 , sharedStatePaymentScript = template pTemplate
+                 , sharedStateDelegationScript = template <$> dTemplateM
+                 , sharedStateDerivationPrefix = prefix
+                 }
+         insertCosigner cs cred = do
+             deleteWhere [CosignerKeyWalletId ==. wid]
+
+             dbChunked insertMany_
+                 [ CosignerKey wid cred (hex $ xpubToBytes xpub) c
```

Comment:
What about sticking to `serializeXPub` from the `PersistPublicKey` abstraction?

Reply:
yes, I was thinking about this. Because I use types from cardano-address, I use `Map Cosigner XPub` there. So I can use `serializeXPub` but before that I would need to lift `XPub` to `key XPub` .  But for being consistent maybe it is good idea. Adpoted